src/e2e-test/features/spanner/source/SpannerToBigQuery_WithMacro.feature

```diff
@@ -0,0 +1,65 @@
+    Then Enter BigQuery sink property "truncateTable" as macro argument "bqTruncateTable"
+    Then Enter BigQuery sink property "updateTableSchema" as macro argument "bqUpdateTableSchema"
+    Then Validate "BigQuery" plugin properties
+    Then Close the BigQuery properties
+    Then Connect source as "Spanner" and sink as "BigQuery" to establish connection
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Enter runtime argument value "projectId" for key "spannerProjectId"
+    Then Enter runtime argument value "serviceAccountAutoDetect" for key "spannerServiceAccount"
+    Then Enter runtime argument value for Spanner Instance ID key "spannerInstanceId"
+    Then Enter runtime argument value for Spanner Database Name key "spannerDatabaseName"
+    Then Enter runtime argument value for Spanner Source Table Name key "spannerTablename"
+    Then Enter runtime argument value "projectId" for key "bqProjectId"
+    Then Enter runtime argument value "projectId" for key "bqDatasetProjectId"
+    Then Enter runtime argument value "serviceAccountAutoDetect" for key "bqServiceAccount"
+    Then Enter runtime argument value "dataset" for key "bqDataset"
+    Then Enter runtime argument value for BigQuery sink table name key "bqTargetTable"
+    Then Enter runtime argument value "bqTruncateTableTrue" for key "bqTruncateTable"
+    Then Enter runtime argument value "bqUpdateTableSchemaTrue" for key "bqUpdateTableSchema"
+    Then Run the preview of pipeline with runtime arguments
+    Then Verify the preview of pipeline is "success"
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "projectId" for key "spannerProjectId"
+    Then Enter runtime argument value "serviceAccountAutoDetect" for key "spannerServiceAccount"
+    Then Enter runtime argument value for Spanner Instance ID key "spannerInstanceId"
+    Then Enter runtime argument value for Spanner Database Name key "spannerDatabaseName"
+    Then Enter runtime argument value for Spanner Source Table Name key "spannerTablename"
+    Then Enter runtime argument value "projectId" for key "bqProjectId"
+    Then Enter runtime argument value "projectId" for key "bqDatasetProjectId"
+    Then Enter runtime argument value "serviceAccountAutoDetect" for key "bqServiceAccount"
+    Then Enter runtime argument value "dataset" for key "bqDataset"
+    Then Enter runtime argument value for BigQuery sink table name key "bqTargetTable"
+    Then Enter runtime argument value "bqTruncateTableTrue" for key "bqTruncateTable"
+    Then Enter runtime argument value "bqUpdateTableSchemaTrue" for key "bqUpdateTableSchema"
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Get count of no of records transferred to target BigQuery Table
```

Comment:
Replace `Then Get count of no of records transferred to target BigQuery Table `-> `Then Validate records transferred to target BigQuery table with record counts of spanner table`